### PR TITLE
external/dhcpd_lwip: Fix pthread_create return check in dhcps

### PR DESCRIPTION
**Pull request by j11591**
Fix pthread_create return check in dhcps and add detach

### CODE REVIEW

**Review by jeongchanKim — APPROVED** [review]
LGTM

**Review by sunghan-chang — APPROVED** [review]

**Review by thapav — APPROVED** [review]
LGTM

File: external/dhcpd/dhcpd_lwip.c [hunk 1]
```diff
@@ -332,13 +332,14 @@ int dhcp_server_start(char *intf, dhcp_sta_joined_cb dhcp_join_cb)
...
 	ret = pthread_create(&g_dhcpd_tid, &attr, _dhcpd_join_handler, (void *)data);
-	if (ret < 0) {
+	if (ret != OK) {
 		free(data->intf);
 		free(data);
 		ndbg("dhcps create dhcp handler fail(%d) errno %d\n", ret, errno);
 		return ERROR;
 	}
 	pthread_setname_np(g_dhcpd_tid, "dhcpd cb handler");
+	pthread_detach(g_dhcpd_tid);
```
Comment (pillip8282) [inline external/dhcpd/dhcpd_lwip.c:342]: You don't need to call pthread_detach here. because pthread_join is called in dhcps_stop().

**Review by pillip8282 — CHANGES_REQUESTED** [review]

**Review by jeongchanKim — APPROVED** [review]
LGTM

**Review by pillip8282 — APPROVED** [review]
Lgtm

**Pull request merged**